File: src/main/java/cloud/FileUtil.java
package cloud;

import java.io.File;
import java.util.List;

public final class FileUtil {

	public static String getWholePath(List<String> paths) {
		return getWholePath(paths.toArray(new String[paths.size()]));
	}
	
	public static String getWholePath(String... paths) {

		StringBuilder wholePath = new StringBuilder("");

		for (int i = 0; i < paths.length; i++) {
			if (i > 0) {
				wholePath.append(File.separator);
			}
			wholePath.append(paths[i]);
		}

		return wholePath.toString();
	}

	public static void deleteFile(File file) {

		if (file.isFile()) {
			file.delete();
			return;
		}

		File[] files = file.listFiles();

		for (int i = 0; i < files.length; i++) {
			deleteFile(files[i]);
		}
		file.delete();
	}
}

File: src/main/kotlin/cloud/util/FileUtil.kt
package cloud.util

import java.io.File

object FileUtil {
    fun getWholePath(paths: List<String>): String {
        return getWholePath(*paths.toTypedArray())
    }

    fun getWholePath(vararg paths: String): String {
        val wholePath = StringBuilder("")
        for (i in paths.indices) {
            if (i > 0) {
                wholePath.append(File.separator)
            }
            wholePath.append(paths[i])
        }
        return wholePath.toString()
    }

    fun deleteFile(file: File) {
        if (file.isFile) {
            file.delete()
            return
        }
        val files = file.listFiles()
        for (i in files.indices) {
            deleteFile(files[i])
        }
        file.delete()
    }

    fun getRelativePath(file: File,rootFile:File):List<String>{
        //相对路径
        val path = mutableListOf<String>()

        var parent = file
        while (parent != rootFile) {
            path.add(0,parent.name)
            parent = parent.parentFile
        }

        return path
    }
}